Framework/Algorithms/src/PDFFourierTransform2.cpp

```diff
@@ -399,11 +419,15 @@ void PDFFourierTransform2::exec() {
     */
   }
 
+  double rho0 = determineRho0();
+  const Kernel::Material &material = inputWS->sample().getMaterial();
+  const auto cohScatLen = material.cohScatterLength();
```

Comment:
I think that if the cohScatLen is zero and PDFType=G_k here we should throw an exception with a message saying "Coherent Scattering Length is zero. Please check a sample material has been specified" or similar.
This will involve you storing getProperty("PDFType") in a variable as well I think
You would get this error condition if you ran this script which is a variation on the usage example in the algorithm documentation. At the moment the algorithm just runs with cohScatLen=0 and produces a zero output:
```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

# Simulates Load of a workspace with all necessary parameters
import numpy as np;
xx = np.array(range(0,100))*0.1
yy = np.exp(-(2.0 * xx)**2)
yy = np.delete(yy,-1) # need one less Y value than X value for histogram data
ws = CreateWorkspace(DataX=xx, DataY=yy, UnitX='MomentumTransfer')
Rt = PDFFourierTransform(ws, SofQType='S(Q)-1', PDFType='G_k(r)')

# Look at sample results:
print('part of S(Q)-1 and its correlation function')
for i in range(10):
   print('! {0:4.2f} ! {1:5f} ! {2:f} ! {3:5f} !'.format(xx[i], yy[i], Rt.readX(0)[i], Rt.readY(0)[i]))
```